mars/services/cluster/supervisor/node_info.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 1999-2020 Alibaba Group Holding Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import time
+from collections import defaultdict
+from typing import Dict, List
+
+from .... import oscar as mo
+
+DEFAULT_NODE_DEAD_TIMEOUT = 120
+DEFAULT_NODE_CHECK_INTERVAL = 1
+
+
+class NodeInfoCollectorActor(mo.Actor):
+    def __init__(self, timeout=None, check_interval=None):
+        self._cat_to_nodes = defaultdict(set)
```

Comment:
I tried to understand what `category` is, to my knowledge, it means `supervisor`, `worker`? How about adding an Enum to make it more clear? and maybe it could be renamed to `Role`?